src/deluge/io/midi/midi_engine.cpp

```diff
@@ -82,8 +82,24 @@ void usbSendComplete(int ip) {
 	bool inHostMode = (g_usb_usbmode == USB_HOST);
 
 	// If in host mode, see if we want to send to another device
+	// NB: this only gets called in host mode anyway, refactor!
 	if (inHostMode) {
 
+		// check if there was more to send on the same device
+		bool has_more = connectedDevice->consumeBytes();
+		if (has_more) {
```

Comment:
Should this modify flushUSBMidiToHostedDevice instead of being added inline?

Reply:
yes, I will do so